src/Core/src/Platform/iOS/KeyboardAutoManagerScroll.cs

```diff
@@ -659,6 +657,42 @@ static void AnimateRootView(CGRect rect)
 			ContainerView.Frame = rect;
 	}
 
+	static void ApplyContentInset(UIScrollView? scrolledView, UIScrollView? lastScrollView)
+	{
+		if (scrolledView is null || lastScrollView is null || ContainerView is null)
+			return;
+
+		var keyboardIntersect = CGRect.Intersect(KeyboardFrame, scrolledView.Frame);
+		nfloat movedContainerDistance = 0;
+
+		// if we are moving or have moved the containerview frame, we need to account for that
+		if (TopViewBeginOrigin != InvalidPoint && ContainerView is not null)
+		{
+			movedContainerDistance = TopViewBeginOrigin.Y - ContainerView.Frame.Y;
+		}
+
+		var frameInWindow = ContainerView!.ConvertRectToView(scrolledView.Frame, null);
+		keyboardIntersect = CGRect.Intersect(KeyboardFrame, frameInWindow);
+
+		var bottomInset = keyboardIntersect.Height;
+		var bottomScrollIndicatorInset = bottomInset;
+
+		bottomInset = nfloat.Max(StartingContentInsets.Bottom, bottomInset);
+		bottomScrollIndicatorInset = nfloat.Max(StartingScrollIndicatorInsets.Bottom, bottomScrollIndicatorInset);
+
+		if (OperatingSystem.IsIOSVersionAtLeast(11, 0))
+		{
+			bottomInset -= scrolledView.SafeAreaInsets.Bottom;
+			bottomScrollIndicatorInset -= scrolledView.SafeAreaInsets.Bottom;
+		}
+
+		var movedInsets = scrolledView.ContentInset;
+		movedInsets.Bottom = bottomInset;
+
+		if (lastScrollView.ContentInset != movedInsets)
+			UIView.Animate(AnimationDuration, 0, UIViewAnimationOptions.CurveEaseOut, () => AnimateInset(scrolledView, movedInsets, bottomScrollIndicatorInset), () => { });
+	}
+
 	static UIScrollView? FindParentScroll(UIScrollView? view)
 	{
 		while (view is not null)
```

Comment:
I cannot comment under this line, but there's another issue here while checking `view.ScrollEnabled`.
We should consider only views with **vertical** scroll enabled, therefore we should exclude horizontal collection views from `FindParentScroll`:
```cs
if (view.ScrollEnabled && !IsHorizontalCollectionView(view)) {
    return view;
}
```

```cs
private static bool IsHorizontalCollectionView(UIView collectionView)
    => collectionView is UICollectionView { CollectionViewLayout: UICollectionViewFlowLayout { ScrollDirection: UICollectionViewScrollDirection.Horizontal }};
```

You'll notice that if you do something like this in a page, the content is messed up when the entry is focused.
```
Grid RowDefinitions="*,Auto"
- Border
- CollectionView HeightRequest="64" ItemsLayout={LinearItemsLayout Orientation="Horizontal"}
  - ItemTemplate => Entry
```